### PR TITLE
Edge cases and refactor

### PR DESCRIPTION
**Pull request by GrayMyers**
This branch adds a clamp method which is used to restrict a number between a min and max value.  This is currently used for edge case handling in board generation.  Also moved adjacent cell hash creation into its own method, as it will be needed to implement "smart" guessing for iteration 4.  Added tests for clamp method.  All tests passing.

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -46,14 +59,17 @@ def consecutive?(coords)
...
       left: cell[0] + (cell[1].to_i - 1).to_s,
       right: cell[0] + (cell[1].to_i + 1).to_s
-      }
-    adjacent_cells.key(cell2)
+    }
+  end
+
+  def check_adjacent(cell,cell2)
+    adjacent_cells(cell).key(cell2)
   end
```
Comment (cunninghamge) [inline lib/board.rb:73]: I agree that this adjacent_cells method will make things easier later on. Since the new check_adjacent method is just a helper method for consecutive?, and we are passing check_adjacent two values just to pass those same two values to adjacent_cells, do we want to fold that action into the consecutive method and get rid of check_adjacent? We could just change the results block in consecutive to read `adjacent_cells(pair[0]).key(pair[1])`

File: lib/board.rb [hunk 2]
```diff
@@ -10,7 +10,20 @@ def initialize(width = 4,height = 4)
     @height = height
   end
 
+  def clamp(num,min,max)
+    result = [min,num,max].sort[1]
+    if result != num
+      puts "Warning: Number clamped to value #{result}"
```
Comment (cunninghamge) [inline lib/board.rb:16]: Will this message be printed to the user if they try to create a board size outside of our parameters? If so we might want to make it sound a little less technical.

File: lib/board.rb [hunk 3]
```diff
@@ -10,7 +10,20 @@ def initialize(width = 4,height = 4)
...
+    end
+    result
+  end
+
   def create_empty_cells(width,height)
+    default_min = 2 #there is no reason to work with a board smaller than this
+    default_y_max = 26 #to prevent invalid characters
+    default_x_max = 10 #to prevent row misalignment
```
Comment (cunninghamge) [inline lib/board.rb:24]: If we really wanted to we could probably get around this by adding an extra space in the rendering if the column number is greater than 9, but admittedly it does not look as neat.

File: lib/board.rb (hunk 1 above)
Comment (GrayMyers) [reply lib/board.rb]: The check_adjacent method can be absorbed back into the check_adjacent method, however ``adjacent_cells(pair[0]).key(pair[1])`` isn't the most readable, so that logic can be broken up to improve readability.

File: lib/board.rb (hunk 2 above)
Comment (GrayMyers) [reply lib/board.rb]: Yeah, having a static warning message which is extremely vague is not a good idea, I will refactor this.

File: lib/board.rb (hunk 3 above)
Comment (GrayMyers) [reply lib/board.rb:22]: Adjusting spacing for multi digit numbers would be rather awkward to implement and the board wouldn't look good.  On top of this, a board larger than 10 squares would probably cause the game to be very tedious.  I like the idea of keeping the cap, but the warning message should become dynamic so that the clamp function can be more user friendly.
